File: 2519-find-the-original-array-of-prefix-xor/2519-find-the-original-array-of-prefix-xor.cpp
class Solution {
public:
    std::vector<int> findArray(std::vector<int>& pref) {
        std::vector<int> ans(pref.size());
        ans[0] = pref[0];
        for (int i = 1; i < ans.size(); i++) {
            ans[i] = pref[i] ^ pref[i - 1];
        }
        return ans;
    }
};

File: 2519-find-the-original-array-of-prefix-xor/2519-find-the-original-array-of-prefix-xor.js
/**
 * @param {number[]} pref
 * @return {number[]}
 */
var findArray = function(pref) {
    let ans = new Array(pref.length);
    ans[0] = pref[0];
    for (let i = 1; i < ans.length; i++) {
        ans[i] = pref[i] ^ pref[i - 1];
    }
    return ans;
};
